$MirOS: ports/print/cups/patches/patch-berkeley_lpq_c,v 1.1.7.1 2005/03/18 15:51:00 tg Exp $
--- berkeley/lpq.c.orig	Tue Mar 28 15:59:54 2006
+++ berkeley/lpq.c	Tue May  9 17:39:25 2006
@@ -385,9 +385,9 @@ show_jobs(const char *command,		/* I - C
   if (dest == NULL)
   {
     if (id)
-      sprintf(resource, "ipp://localhost/jobs/%d", id);
+      snprintf(resource, sizeof(resource), "ipp://localhost/jobs/%d", id);
     else
-      strcpy(resource, "ipp://localhost/jobs");
+      strlcpy(resource, "ipp://localhost/jobs", sizeof(resource));
 
     ippAddString(request, IPP_TAG_OPERATION, IPP_TAG_URI, "job-uri",
                  NULL, resource);
